packages/core/admin/admin/src/content-manager/components/RelationInputDataManager/utils/select.js

```diff
@@ -40,23 +46,35 @@ function useSelect({ isUserAllowedToEditField, isUserAllowedToReadField, name, q
     return allowedFields.includes(name);
   }, [isCreatingEntry, isUserAllowedToReadField, name, readActionAllowedFields]);
 
-  // /content-manager/[collection-type]/[content-type]/[id]/[field-name]
+  // /content-manager/relations/[model]/[id]/[field-name]
   const relationFetchEndpoint = useMemo(() => {
-    const collectionTypePrefix = isSingleType ? 'single-types' : 'collection-types';
-
     if (isCreatingEntry) {
       return null;
     }
 
-    return getRequestUrl(
-      `${collectionTypePrefix}/${slug}/${initialData.id}/${name.split('.').at(-1)}`
-    );
-  }, [isCreatingEntry, slug, initialData, name, isSingleType]);
+    if (componentUid) {
+      const fieldNameKeys = name.split('.');
+      const componentId = get(initialData, fieldNameKeys.slice(0, -1))?.id;
```

Comment:
As far as I understand, this hardcodes three levels of nesting. I think the CM overall avoids that and should work with any level of nesting theoretically. I might be wrong, so this is a question for now :)

Reply:
> this hardcodes three levels of nesting.

oh 🤔 the admin doesn't seem to let me nest them more than 2 - is it something that we support? or maybe only through file config?
does that mean if I manage to config 3-4 nested level, the name won't contain all fields name to access the id in `initialData`?

<img width="983" alt="image" src="https://user-images.githubusercontent.com/71838159/194242038-935480ea-5c69-4045-b533-1fee7f1c5a79.png">
<img width="212" alt="image" src="https://user-images.githubusercontent.com/71838159/194242073-4fb5ef85-824c-4e31-8a5d-acb780e79416.png">

> I would pass down the componentId as a prop

mh I didn't find another place than by using `initialData` to get `componentId` - did I miss a place where the data holds this information?

Reply:
@gu-stav update, I tested by creating manually the nested components and it seems to work, were you referring to something else? 👀

<img width="832" alt="image" src="https://user-images.githubusercontent.com/71838159/194246370-42abe32f-c076-4d04-90fd-2904697308ef.png">